### ISSUE TITLE
document can give a misleading error message

### ISSUE THREAD

**Issue by ggrothendieck** [r-lib/roxygen2#1088]
When running document() one can get an error about missing dependency when that dependency is not missing. It seems that it can be due to a faulty .onLoad, a faulty Namespace file (which perhaps was not updated from an older version) or other problems.  If "X" is the package that is reported to be missing then the output of requireNamespace("X") is generally more useful in diagnosing this than the message that comes from devtools which can be incorrect and it would be good if that message were output.

**Comment by hadley** [comment]
Do you have more details? There are many things that might prevent a package from successfully loading.

**Comment by ggrothendieck** [comment]
I already mentioned this but the message that `requireNamespace` gives is more informative so if you knew to run that you would get a clue to what is the wrong.  A more accurate error message would be that 

```
requireNamespace("X") failed with message "..." .   Possible causes are a missing package, a faulty .onLoad, an erroneous NAMESPACE file or other.
```

When I encountered this I tracked it down to requireNamespace in document and once I ran that and saw the output it gave me a good clue as to what to do.  It seems this course of action has been reinvented a number of times as we see from this stackoverflow answer where the writer did the same thing.  

https://stackoverflow.com/questions/33428055/dependency-package-package-name-not-available/47104752#47104752

**Comment by hadley** [comment]
Thanks for filing this bug report! Unfortunately because it seems unlikely to affect many people, we don't have the development resources to fix at this time. It's our policy to close such issues to help stay focussed on the biggest problems, but the issue is still indexed by google, so if other people hit it, they'll be able to find it, and we can consider reopen it if it turns out to be a common problem. Thanks for reporting and I'm sorry we couldn't help more 😞.
